gateway/modules/eventing/helpers.go

```diff
@@ -85,10 +90,14 @@ func (m *Module) batchRequests(ctx context.Context, requests []*model.QueueEvent
 	return nil
 }
 
-func (m *Module) generateQueueEventRequest(token int, name string, batchID, status string, event *model.QueueEventRequest) *model.EventDocument {
+func (m *Module) generateQueueEventRequest(token int, name, eventDocID, batchID, status string, event *model.QueueEventRequest) *model.EventDocument {
```

Comment:
Instead of changing the definition of this function, make another function `generateQueueEventRequestRaw` with the same body of the current function. `generateQueueEventRequest` will simply call `generateQueueEventRequestRaw` with an empty `eventDocID`. This way the functions currently using `generateQueueEventRequest` will not have to be changed.

You will need to test `generateQueueEventRequest`

Reply:
I have already made changes to the functions calling these functions and written tests for them as well

Reply:
And it will create unpredictable field in the functions calling the original function ```generateQueueEventRequest```

Reply:
I'm sorry test the raw function my bad